include/connection.h

```diff
@@ -50,10 +49,26 @@ typedef struct NodeConnectionEntry
 } NodeConnectionEntry;
 
 
+
+typedef struct {
+	ShardId shardId;
+	int replicaCount;
+	bool* status;
```

Comment:
if we're keeping it as a bool, can we rename it to "sentPrepare"? That would make its purpose more clear (also applies to functions where its passed as a parameter).